Session/Home/GlobalSearch.swift

```diff
@@ -0,0 +1,342 @@
+            cell.threadViewModel = searchResult?.thread
+            cell.configure(messageDate: searchResult?.messageDate, snippet: searchResult?.snippet, searchText: searchResultSet.searchText)
+            return cell
+        case .messages:
+            let sectionResults = searchResultSet.messages
+            let cell = tableView.dequeueReusableCell(withIdentifier: ConversationCell.reuseIdentifier) as! ConversationCell
+            cell.isShowingGlobalSearchResult = true
+            let searchResult = sectionResults[safe: indexPath.row]
+            cell.threadViewModel = searchResult?.thread
+            cell.configure(messageDate: searchResult?.messageDate, snippet: searchResult?.snippet, searchText: searchResultSet.searchText)
+            return cell
+        }
+    }
+}
+
+// MARK: - UIScrollViewDelegate
+
+extension GlobalSearchViewController {
+    public override func scrollViewWillBeginDragging(_ scrollView: UIScrollView) {
+        delegate?.globalSearchViewWillBeginDragging()
+    }
+}
+
+// MARK: -
+
+class EmptySearchResultCell: UITableViewCell {
+    static let reuseIdentifier = "EmptySearchResultCell"
+
+    private lazy var messageLabel: UILabel = {
+        let result = UILabel()
+        result.textAlignment = .center
+        result.numberOfLines = 3
+        result.textColor = Colors.text
+        result.text = NSLocalizedString("CONVERSATION_SEARCH_NO_RESULTS", comment: "")
+        return result
+    }()
+    
+    private lazy var spinner: NVActivityIndicatorView = {
+        let result = NVActivityIndicatorView(frame: CGRect.zero, type: .circleStrokeSpin, color: Colors.text, padding: nil)
+        result.set(.width, to: 40)
+        result.set(.height, to: 40)
+        return result
+    }()
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        
+        contentView.addSubview(messageLabel)
+        messageLabel.autoSetDimension(.height, toSize: 150)
+        messageLabel.autoPinEdge(toSuperviewMargin: .top, relation: .greaterThanOrEqual)
+        messageLabel.autoPinEdge(toSuperviewMargin: .leading, relation: .greaterThanOrEqual)
+        messageLabel.autoPinEdge(toSuperviewMargin: .bottom, relation: .greaterThanOrEqual)
+        messageLabel.autoPinEdge(toSuperviewMargin: .trailing, relation: .greaterThanOrEqual)
+        messageLabel.autoVCenterInSuperview()
+        messageLabel.autoHCenterInSuperview()
+        messageLabel.setContentHuggingHigh()
+        messageLabel.setCompressionResistanceHigh()
+
+        contentView.addSubview(spinner)
+        spinner.autoCenterInSuperview()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        notImplemented()
+    }
+
+    public func configure(searchText: String) {
+        if searchText.count < GlobalSearchViewController.minimumSearchTextLength {
+            spinner.stopAnimating()
+            spinner.startAnimating()
```

Comment:
Should this just have the `startAnimating` call?

Reply:
This is a weird situation. The spinner won't change the colour as the theme changed. I think it is because the animation is on a layer. The `stopAnimating()` can remove the layer and then `startAnimating()` can recreate a layer with the correct colour. I'll add some comment in the code.